packages/ripple-binary-codec/test/hash.test.js

```diff
@@ -77,6 +78,12 @@ describe('Currency', function () {
     const currency = '015841551A748AD2C1F76FF6ECB0CCCD00000000'
     expect(Currency.from(currency).toJSON()).toBe(currency)
   })
+
+  test('Can handle other non-standard currency codes', () => {
+    const currency = '0000000000414C6F676F30330000000000000000'
```

Comment:
This is not quite right - For non-standard currency codes the 1st byte CANNOT be 0x00 - https://xrpl.org/currency-formats.html#nonstandard-currency-codes

Reply:
People are doing it anyways and we have no means of stopping them

Reply:
The encoding for non non-standard currency codes is whatever rippled says it is

In this case, rippled says that this should work like this, so it should work like this.

Because rippled doesnt actually enforce how non-standard currency codes work, we cant enforce how this works.